internal/pkg/service/objectstorage/cloud/azure/service.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/object-storage-operator/internal/pkg/cluster"
+	"github.com/giantswarm/object-storage-operator/internal/pkg/service/objectstorage"
+)
+
+type AzureObjectStorageService struct {
+}
+
+func (s AzureObjectStorageService) NewAccessRoleService(ctx context.Context, logger logr.Logger, cluster cluster.Cluster) (objectstorage.AccessRoleService, error) {
+	//TODO
+	return nil, nil
+}
+
+func (s AzureObjectStorageService) NewObjectStorageService(ctx context.Context, logger logr.Logger, cluster cluster.Cluster, cli client.Client) (objectstorage.ObjectStorageService, error) {
+	clust := &unstructured.Unstructured{}
+	clust.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
Why do you need that and thé client when it should bé in thé clustergetter implémentation ?

Reply:
I refactored a bit